apidocs.yaml

```diff
@@ -0,0 +1,826 @@
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/AddedAttachments"
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/{id}/attachments/{attachmentId}:
+    get:
+      summary: Retrieve attachment from application
+      operationId: retrieve_attachment_from_application_v1
+      tags:
+        - "attachments"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to retrieve attachment from
+          required: true
+          schema:
+            type: string
+        - name: attachmentId
+          in: path
+          description: ID of attachment to retrieve
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Successful Response
+          content:
+            application/pdf:
+              schema:
+                type: string
+                format: binary
+  /api/v1/applications/{id}/remove-attachment:
+    post:
+      summary: Remove attachment from application
+      operationId: remove_attachment_from_application_v1
+      tags:
+        - "attachments"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to remove attachment from
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/DeleteAttachment"
+      responses:
+        "204":
+          description: Successful Response (no content)
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/{id}/add-member:
+    post:
+      summary: Add member to application
+      operationId: add_member_to_application_v1
+      tags:
+        - "members"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to add member to
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: Successful Response (no content)
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/{id}/remove-member:
+    post:
+      summary: Remove
+      operationId: remove_member_from_application_v1
+      tags:
+        - "members"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to remove member from
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RemoveMember"
+      responses:
+        "204":
+          description: Successful Response (no content)
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/{id}/manage-datasets:
```

Comment:
I am not sure to understand the action implemented by this endpoint, especially the meaning of "manage". 

Does that mean adding (linking) datasets to a certain application?

Reply:
For my own understanding, can you explain me why you did so instead of creating multiple endpoints with specific http verbs for adding, deleting and updating datasets? 
Like you did for /api/v1/applications/{id}/remove-member and /api/v1/applications/{id}/add-member

It is because we are dealing with a collection here? I must admit, that would become cumbersome if we had to do 
DELETE /api/v1/applications/{id}/{dataset_id}

Reply:
I will close this conversation to keep this topic in a single thread.

https://github.com/GenomicDataInfrastructure/gdi-userportal-access-management-service/pull/14#discussion_r1484338971